tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,303 @@
+	namespace                       = "namespace"
+	name                            = "name"
+	clusterName                     = "clusterName"
+	userPrincipalName               = "userPrincipalName"
+	crtbNameWithTenChar             = 10
+	ID                              = "id"
+	crtbConflictError               = "409 Conflict"
+	crtbAlreadyExisted              = "AlreadyExists"
+	badRequest                      = "400 Bad Request"
+	notFound                        = "404 Not Found"
+	cluserNameAndNameSpaceSameValue = "Forbidden: clusterName and namespace must be the same value"
+)
+
+func (crtb *CRTBGenTestSuite) TearDownSuite() {
+	crtb.session.Cleanup()
+}
+
+func (crtb *CRTBGenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	crtb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(crtb.T(), err)
+
+	crtb.client = client
+	crtb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crtb.T(), crtb.clusterName, "Cluster name to install should be set")
+
+	crtb.clusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.clusterName)
+	require.NoError(crtb.T(), err, "Error getting cluster ID")
+
+	crtb.crtbJsonTemplateData = setupJsonTemplateForClusterRoleTemplateBinding().crtbJsonTemplateData
+	require.NoError(crtb.T(), err)
+
+	crtb.newUser, err = users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(crtb.T(), err)
+}
+
+func setupJsonTemplateForClusterRoleTemplateBinding() CRTBGenTestSuite {
+	crtbTemplate := v3.ClusterRoleTemplateBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiVersion,
+			Kind:       kind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		ClusterName:      clusterName,
+		RoleTemplateName: roleTemplateName,
+		UserName:         userName,
+	}
+
+	jsonTemplateData, _ := json.Marshal(crtbTemplate)
+
+	return CRTBGenTestSuite{crtbJsonTemplateData: jsonTemplateData}
+}
+
+func unmarshalCrtbJsonTemplate(t *testing.T, readJson []byte, clusterID string, crtbName string, newUser string, roleTemplateId string) map[string]interface{} {
+	var inputData map[string]interface{}
+	err := json.Unmarshal(readJson, &inputData)
+	require.NoError(t, err)
+
+	inputData[metadata].(map[string]interface{})[namespace] = clusterID
+	inputData[metadata].(map[string]interface{})[name] = crtbName
+	inputData[clusterName] = clusterID
+	inputData[userName] = newUser
+	inputData[roleTemplateName] = roleTemplateId
+	return inputData
+}
+
+func createCrtbWithAllRequiredInputValues(t *testing.T, client *rancher.Client, clusterID string, newUser *management.User, roleTemplateName string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(crtbNameWithTenChar)
+
+	inputData := unmarshalCrtbJsonTemplate(t, readJson, clusterID, crtbName, newUser.ID, roleTemplateName)
+	require.NoError(t, err)
```

Comment:
This is still not resolved, Das. **unmarshalReadJson** function does not return an error that you can assert here